### ISSUE TITLE
Crash on Google Drive File Selection

### ISSUE THREAD

**Issue by i140394** [HBiSoft/PickiT#14]
My App is crashing when I have selected a file from Google Drive. 
PickiT: 0.1.9
Mobile: Android Version: 6.0

![Screen Shot 2020-02-21 at 4 35 27 PM](https://user-images.githubusercontent.com/32557545/75029543-3a81d100-54c8-11ea-8632-335aedc14358.png)

**Comment by HBiSoft** [comment]
What version are you using?

**Comment by i140394** [comment]
Library: 0.1.9
OS: Android 6.0

**Comment by HBiSoft** [comment]
Can you send me a link to a file I can test with? The size should not be zero.

**Comment by i140394** [comment]
Sure.

**Comment by i140394** [comment]
It's only happening when I am selecting drive file from Recents section.

**Comment by i140394** [comment]
File Link: https://drive.google.com/open?id=1uNzJ5p-ZpNs0Ii-Yc5zEpYAwWgvJb5PGzcOJajLxTNc

**Comment by HBiSoft** [comment]
Can you please share the uri when selecting it from Google Drive and when you select it from Recents?

**Comment by i140394** [comment]
Ok. It is crashing for Google sheet/doc etc files from **Recents** and **Drive** also.
I will share you the Uris shortly.

**Comment by i140394** [comment]
For your reference: I think this is the same issue as below one
[https://github.com/HBiSoft/PickiT/issues/9#issue-525143398](url)

**Comment by i140394** [comment]
From Drive: **content://com.google.android.apps.docs.storage/document/acc%3D9%3Bdoc%3Dencoded%3DJnO56ZsvMKX3juRDYo4DF7Xyod10TSNi6bERrFPTBkeJeXMci%2FlQaQ%3D%3D**
From Recents:
**content://com.google.android.apps.docs.storage/document/acc%3D9%3Bdoc%3Dencoded%3DJnO56ZsvMKX3juRDYo4DF7Xyod10TSNi6bERrFPTBkeJeXMci%2FlQaQ%3D%3D**

**Comment by HBiSoft** [comment]
I tested on my device with the demo application and I do not see the crash, please download the project and change the following:

https://github.com/HBiSoft/PickiT/blob/c3cb420697a04a1f2c3909c0cf9fa9635bf024ef/app/src/main/java/com/hbisoft/pickitexample/MainActivity.java#L92

to

```java
intent.setType("application/pdf");
```

The run that on your device and see if you still get the error.

**Comment by i140394** [comment]
Yes. It is still crashing.

**Comment by HBiSoft** [comment]
It doesn't make sense, here is a screen shot when selecting the file you sent me from Google Drive:

![Screenshot_20200221-145403_PickiT example](https://user-images.githubusercontent.com/35602540/75036278-29ca5e80-54ba-11ea-9800-ac3450123829.jpg)

I will change the code to check if the file size is not 0 (which it should not be, but I will do it anyway).

**Comment by HBiSoft** [comment]
The other thing I do not understand is that your log is pointing to:

https://github.com/HBiSoft/PickiT/blob/c3cb420697a04a1f2c3909c0cf9fa9635bf024ef/pickit/src/main/java/com/hbisoft/pickit/DownloadAsyncTask.java#L98

but the dividing is happening on line 93:

https://github.com/HBiSoft/PickiT/blob/c3cb420697a04a1f2c3909c0cf9fa9635bf024ef/pickit/src/main/java/com/hbisoft/pickit/DownloadAsyncTask.java#L93

**Comment by HBiSoft** [comment]
Please add this:

```java
Log.e("Size =". ""+size);
```
here:

https://github.com/HBiSoft/PickiT/blob/c3cb420697a04a1f2c3909c0cf9fa9635bf024ef/pickit/src/main/java/com/hbisoft/pickit/DownloadAsyncTask.java#L78

and let me know if the size is 0.

**Comment by i140394** [comment]
Hope this will help.
[https://drive.google.com/open?id=171_Kg1UQYlwFKiLYBLz8uGmK9oRh2KG0](url)

**Comment by HBiSoft** [comment]
> Please add this:
> 
> ```java
> Log.e("Size =". ""+size);
> ```
> 
> here:
> 
> https://github.com/HBiSoft/PickiT/blob/c3cb420697a04a1f2c3909c0cf9fa9635bf024ef/pickit/src/main/java/com/hbisoft/pickit/DownloadAsyncTask.java#L78
> 
> and let me know if the size is 0.

Please check the above

**Comment by i140394** [comment]
OK. Sure

**Comment by i140394** [comment]
Yes. Size is zero.
I have added Log statement at line number 79.

![Screen Shot 2020-02-24 at 11 17 18 AM](https://user-images.githubusercontent.com/32557545/75131090-60d98380-56f7-11ea-91e8-32b700737c4b.png)

![Screen Shot 2020-02-24 at 11 17 38 AM](https://user-images.githubusercontent.com/32557545/75131100-6fc03600-56f7-11ea-84d5-8ef25cec80da.png)

**Comment by HBiSoft** [comment]
Thank you.

Please change the entire `while` loop:

https://github.com/HBiSoft/PickiT/blob/c3cb420697a04a1f2c3909c0cf9fa9635bf024ef/pickit/src/main/java/com/hbisoft/pickit/DownloadAsyncTask.java#L89

to the following:

```java
while ((count = bis.read(data)) != -1) {
    if (!isCancelled()) {
        total += count;
        if (size != -1) {
            try {
                publishProgress((int) ((total * 100) / size));
            }catch(Exception e){
                Log.i("PickiT -", "File size is less than 1");
                publishProgress(0);
            }
        }
        fos.write(data, 0, count);
    }
}
```

Let me know if the crash goes away and I will update the library.

**Comment by i140394** [comment]
The crash is gone. But I was getting file size as **0**. Is it because size is **int** field?
And my file size is less than 1MB. Is that it?

**Comment by HBiSoft** [comment]
Yes, the problem is that your file size is `0.23MB` for example, and it gets rounded to 0, thus the error. I'm busy updating the library.

**Comment by i140394** [comment]
OK. Thank You.

**Comment by HBiSoft** [comment]
This is fixed in `0.1.10`. You can test it with the [demo application](https://github.com/HBiSoft/PickiT/releases/download/0.1.10/PickiTDemo.apk) if you would like.

Please test it and close the issue if it is fixed.
